server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/dto/DLPConfigurationDTO.java

```diff
@@ -0,0 +1,140 @@
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.dto;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.james.dlp.api.DLPConfigurationItem;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.github.steveash.guavate.Guavate;
+import com.google.common.base.Preconditions;
+
+public class DLPConfigurationDTO {
+
+    public static List<DLPConfigurationItem> toDLPConfigurations(List<DLPConfigurationDTO> dtos) {
+        Preconditions.checkNotNull(dtos);
+
+        return dtos.stream()
+            .map(DLPConfigurationDTO::toDLPConfiguration)
+            .collect(Guavate.toImmutableList());
+    }
+
+    public static List<DLPConfigurationDTO> toDTOs(List<DLPConfigurationItem> dlpConfigurations) {
+        Preconditions.checkNotNull(dlpConfigurations);
+
+        return dlpConfigurations.stream()
+            .map(DLPConfigurationDTO::toDTO)
+            .collect(Guavate.toImmutableList());
+    }
+
+    private static DLPConfigurationItem toDLPConfiguration(DLPConfigurationDTO dto) {
+        return DLPConfigurationItem.builder()
+            .id(DLPConfigurationItem.Id.of(dto.id))
+            .expression(dto.expression)
+            .explanation(dto.explanation)
+            .targetsSender(dto.targetsSender)
+            .targetsRecipients(dto.targetsRecipients)
+            .targetsContent(dto.targetsContent)
+            .build();
+    }
+
+    private static DLPConfigurationDTO toDTO(DLPConfigurationItem dlpConfiguration) {
+        DLPConfigurationItem.Targets targets = dlpConfiguration.getTargets();
+        return new DLPConfigurationDTO(
+            dlpConfiguration.getId().asString(),
+            dlpConfiguration.getRegexp(),
+            dlpConfiguration.getExplanation(),
+            Optional.of(targets.isSenderTargeted()),
+            Optional.of(targets.isRecipientTargeted()),
+            Optional.of(targets.isContentTargeted()));
+    }
+
+    private final String id;
+    private final String expression;
+    private final Optional<String> explanation;
+    private final Optional<Boolean> targetsSender;
```

Comment:
Are you sure targets can be missing?

IMO it's either true or false but not null....

Reply:
Targets can be absent in request body then `DLPConfigurationDTO`, but will be set to false if null in `DLPConfigurationItem.Builder.buid()`
Then a `DLPConfigurationItem` always has specifed targets.

Reply:
Maybe we need two DTO:

DLPConfigurationItemDTO && UserSpecifiedDLPConfigurationItemDTO

?

Reply:
Ok